internals/overlord/logstate/loki/loki_test.go

```diff
@@ -191,6 +191,65 @@ func (*suite) TestServerTimeout(c *C) {
 	c.Assert(err, ErrorMatches, ".*context deadline exceeded.*")
 }
 
+func (*suite) TestBufferFull(c *C) {
+	client := loki.NewClientWithOptions(
+		&plan.LogTarget{
+			Name:     "tgt1",
+			Location: "fake",
+		},
+		&loki.ClientOptions{
+			MaxRequestEntries: 3,
+		},
+	)
+
+	checkBuffer := func(expected []any) {
+		checkBuffer(c, client, expected)
```

Comment:
I found this name shadowing a bit confusing. You could just inline `checkBuffer` here and capture `c` and `client`, or rename this local one to `check` or similar.